pkg/backup/backupsink/sink_utils.go

```diff
@@ -0,0 +1,214 @@
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/builtins"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/errors"
+)
+
+// ElidedPrefix returns the prefix of the key that is elided by the given mode.
+func ElidedPrefix(key roachpb.Key, mode execinfrapb.ElidePrefix) ([]byte, error) {
+	switch mode {
+	case execinfrapb.ElidePrefix_TenantAndTable:
+		rest, err := keys.StripTablePrefix(key)
+		if err != nil {
+			return nil, err
+		}
+		return key[: len(key)-len(rest) : len(key)-len(rest)], nil
+
+	case execinfrapb.ElidePrefix_Tenant:
+		rest, err := keys.StripTenantPrefix(key)
+		if err != nil {
+			return nil, err
+		}
+		return key[: len(key)-len(rest) : len(key)-len(rest)], nil
+	}
+	return nil, nil
+}
+
+// adjustFileEndKey checks if the export respsonse end key can be used as a
+// split point during restore. If the end key is not splitable (i.e. it splits
+// two column families in the same row), the function will attempt to adjust the
+// endkey to become splitable. The function returns the potentially adjusted
+// end key and whether this end key is mid row/unsplitable (i.e. splits a 2
+// column families or mvcc versions).
+func adjustFileEndKey(endKey, maxPointKey, maxRangeEnd roachpb.Key) (roachpb.Key, bool) {
+	maxKey := maxPointKey
+	if maxKey.Compare(maxRangeEnd) < 0 {
+		maxKey = maxRangeEnd
+	}
+
+	endRowKey, err := keys.EnsureSafeSplitKey(endKey)
+	if err != nil {
+		// If the key does not parse a family key, it must be from reaching the end
+		// of a range and be a range boundary.
+		return endKey, false
+	}
+
+	// If the end key parses as a family key but truncating to the row key does
+	// _not_ produce a row key greater than every key in the file, then one of two
+	// things has happened: we *did* stop at family key mid-row, so we copied some
+	// families after the row key but have more to get in the next file -- so we
+	// must *not* flush now -- or the file ended at a range boundary that _looks_
+	// like a family key due to a numeric suffix, so the (nonsense) truncated key
+	// is now some prefix less than the last copied key. The latter is unfortunate
+	// but should be rare given range-sized export requests.
+	if endRowKey.Compare(maxKey) <= 0 {
+		return endKey, true
+	}
+
+	// If the file end does parse as a family key but the truncated 'row' key is
+	// still above any key in the file, the end key likely came from export's
+	// iteration stopping early and setting the end to the resume key, i.e. the
+	// next real family key. In this case, we are not mid-row, but want to adjust
+	// our span end -- and where we resume the next file -- to be this row key.
+	// Thus return the truncated row key and false.
+	return endRowKey, false
+
+}
+
+func (s *FileSSTSink) copyPointKeys(ctx context.Context, dataSST []byte) (roachpb.Key, error) {
```

Comment:
why do these two methods that hang off FileSSTSink live in sink_utils? To make this pr simpler, perhaps it is worth just throwing all this stuff in file_sst_sink.go . Github may even detect that this is simple move, like it did for `file_sst_sink_test.go`

Reply:
Oh at the time of refactoring I thought I made throw in some more helper functions in here later down the line for backup compactions. Although thinking about it, these are specific to the file sink implementation. Will move it back